### PR TITLE
Bug 1664023 - Add enrollment ping support to Pioneer v2

### PR DESCRIPTION
**Pull request by acmiyaguchi**
[bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1664023)
[client bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1663857)

Pioneer will be sending an enrollment ping to capture explicit consent before general data collection. This is handled like the deletion-request ping, which is special cased so it does not fail due to the lack of properly encrypted content.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
cc @Dexterp37

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java [hunk 1]
```diff
@@ -51,6 +51,7 @@
   public static final String PIONEER_ID = "pioneerId";
   public static final String STUDY_NAME = "studyName";
   public static final String DELETION_REQUEST_SCHEMA_NAME = "deletion-request";
+  public static final String ENROLLMENT_SCHEMA_NAME = "enrollment";
```
Comment (Dexterp37) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java:54]:
```suggestion
  public static final String ENROLLMENT_SCHEMA_NAME = "pioneer-enrollment";
```

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java [hunk 2]
```diff
@@ -118,13 +119,15 @@ private DecryptPioneerPayloads(ValueProvider<String> metadataLocation,
...
       byte[] payloadData;
-      if (payload.get(SCHEMA_NAME).asText().equals(DELETION_REQUEST_SCHEMA_NAME)) {
-        // The deletion-request ping is special cased within the decryption
-        // pipeline. The Pioneer client requires a JWE payload to exist before
-        // it can be sent. The encrypted data is the empty string encoded with a
-        // throwaway key, in order to satisfy these requirements. We only need
-        // the envelope metadata to shred all client data in a study, so the
-        // encrypted data in the deletion-request ping is thrown away.
+      if (payload.get(SCHEMA_NAME).asText().equals(DELETION_REQUEST_SCHEMA_NAME)
+          || payload.get(SCHEMA_NAME).asText().equals(ENROLLMENT_SCHEMA_NAME)) {
```
Comment (Dexterp37) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java:123]: As you can see [here](https://phabricator.services.mozilla.com/D89718) the name of the schema is the same for both the "global consent" and the "study consent". The difference is the schema namespace and the fact that the study id is missing for "global consent". Do you need to account for that?

Comment (Dexterp37) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java:123]: Moreover, do I need to add a new schema [here](https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/master/schemas/pioneer-debug)?

Comment (acmiyaguchi) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java:123]: It doesn't need to be accounted for here, but a new document namespace in mozilla-pipeline-schemas named `pioneer-meta` will need to be created with a `pioneer-meta/pioneer-enrollment/pioneer-enrollment.1.schema.json` schema where it is the empty schema. 

You can add the new schema to the debug study, too.

**Review by Dexterp37 — APPROVED** [review]

File: ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java (hunk 2 above)
Comment (Dexterp37) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java:123]: @acmiyaguchi I created mozilla-services/mozilla-pipeline-schemas#611

File: ingestion-beam/src/test/resources/pioneer/pioneer-enrollment.sample.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "platformVersion": "82.0a1",
+    "xpcomAbi": "x86_64-gcc3",
+    "channel": "default"
+  },
+  "payload": {
+    "encryptedData": "eyJhbGciOiJFQ0RILUVTIiwiZW5jIjoiQTI1NkdDTSIsImVwayI6eyJjcnYiOiJQLTI1NiIsImt0eSI6IkVDIiwieCI6ImdBQzRSVEVRVlo0QURFTm9yako0aEJjVThsYjJrV1dMYUIxOEdDcDZ2bmsiLCJ5IjoiX1k2RkdlamhseGtKMVVMU05XV2ZER3lCS2g0MWtQZmFxY3d1V1hUejljRSJ9fQ..SKCODugev0CJ7WXI.BYs.jJ8Yn_P8GY5ORl3NUsXb3A",
+    "schemaVersion": 1,
+    "schemaName": "pioneer-enrollment",
+    "schemaNamespace": "pioneer-meta",
+    "encryptionKeyId": "debug",
```
Comment (acmiyaguchi) [inline ingestion-beam/src/test/resources/pioneer/pioneer-enrollment.sample.json:22]:
```suggestion
    "encryptionKeyId": "discarded",
```

**Review by akkomar — APPROVED** [review]

**Pull request merged**